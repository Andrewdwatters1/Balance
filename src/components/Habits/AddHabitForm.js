import React, { Component } from 'react';
import axios from 'axios';
import * as moment from 'moment';
import { ToastContainer, ToastStore } from 'react-toasts';

import './Habits.css'

const add = require('../../assets/plus.png')

export default class AddHabitForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      title: '',
      description: '',
      type: '',
    }
  }

  handleTitleInput = (e) => {
    this.setState({
      title: e.target.value
    })
  }
  handleDescriptionInput = (e) => {
    this.setState({
      description: e.target.value
    })
  }
  handleTypeInput = (e) => {
    this.setState({
      type: e.target.value
    })
  }

  addHabit = (e) => {
    e.preventDefault();
    let { title, description, type } = this.state;
    let dateFormatted = `${new Date().getFullYear()}/${new Date().getMonth()}/${new Date().getDate()}/${new Date().getDay()}`
    let date = new moment(new Date()).format('DDD, Y')
    let habit = { userId: this.props.currentUser.id, title, description, dateFormatted, date, type };
    title ? description ?
      axios.post('/api/habits', habit).then(result => {
        ToastStore.success('Nice! Your habit has been added!')
        this.props.addHabitToList(result.data);
        this.props.getTodaysHabits();
        this.setState({
          title: '',
          description: '',
          type: ''
        }, () => {
          this.props.getUpdatedHabitsByUser()
        })
      })
      :
      ToastStore.error('please enter a description for your habit')
      :
      ToastStore.error('please enter a title for your habit')
  }

  render() {
    console.log(this.props);
    return (
      <div style={{height: '100vh', width: '100vw'}}>
        <div className="add-habit">
          <img src={add} onMouseDown={this.props.toggleForm} className="add-habit-close-form" />
          <form onSubmit={this.addHabit} className="add-habit-form">
            <input onChange={this.handleTitleInput} placeholder="TITLE" value={this.state.title} maxLength="18" />
            <textarea onChange={this.handleDescriptionInput} type="submit text" placeholder='Description' value={this.state.description} rows="4" cols="30"><input /></textarea>
            Category: <select value={this.state.type} onChange={this.handleTypeInput} type="submit" className="add-habit-form-select form-control">
              <option value="Health/Fitness">Health/Fitness</option>
              <option value="Professional">Professional</option>
              <option value="Personal">Personal</option>
            </select>
            <button type="submit">Add Habit</button>
          </form>
        </div>
        <ToastContainer store={ToastStore} position={ToastContainer.POSITION.BOTTOM_RIGHT} />
      </div>

    )
  }
}