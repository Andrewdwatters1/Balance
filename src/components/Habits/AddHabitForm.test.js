import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import AddHabitForm from './AddHabitForm';

Enzyme.configure({ adapter: new Adapter() })

describe('verify environment setup', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AddHabitForm />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('should render without props', () => {
    const app = shallow(<AddHabitForm />);
    expect(app).toMatchSnapshot();
  });
  it('should render child components', () => {
    const app = shallow(<AddHabitForm />);
    expect(app.exists()).toBe(true);
    expect(app.children().length).toBe(2);
  });
});


