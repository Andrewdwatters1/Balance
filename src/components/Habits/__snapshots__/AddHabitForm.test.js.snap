// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verify environment setup should render without props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddHabitForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="add-habit"
>
          <img
                    className="add-habit-close-form"
                    onMouseDown={undefined}
                    src="plus.png"
          />
          <form
                    className="add-habit-form"
                    onSubmit={[Function]}
          >
                    <input
                              maxLength="18"
                              onChange={[Function]}
                              placeholder="TITLE"
                              value=""
                    />
                    <textarea
                              cols="30"
                              onChange={[Function]}
                              placeholder="Description"
                              rows="4"
                              type="submit text"
                              value=""
                    >
                              <input />
                    </textarea>
                    Category: 
                    <select
                              className="add-habit-form-select form-control"
                              onChange={[Function]}
                              type="submit"
                              value=""
                    >
                              <option
                                        value="Health/Fitness"
                              >
                                        Health/Fitness
                              </option>
                              <option
                                        value="Professional"
                              >
                                        Professional
                              </option>
                              <option
                                        value="Personal"
                              >
                                        Personal
                              </option>
                    </select>
                    <button
                              type="submit"
                    >
                              Add Habit
                    </button>
          </form>
</div>,
        <Container
          position="bottom_right"
          store={
                    Object {
                              "data": Object {
                                "action": "",
                                "message": "",
                              },
                              "error": [Function],
                              "info": [Function],
                              "success": [Function],
                              "unwatch": [Function],
                              "warning": [Function],
                              "watch": [Function],
                            }
          }
/>,
      ],
      "style": Object {
        "height": "100vh",
        "width": "100vw",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              className="add-habit-close-form"
              onMouseDown={undefined}
              src="plus.png"
/>,
            <form
              className="add-habit-form"
              onSubmit={[Function]}
>
              <input
                            maxLength="18"
                            onChange={[Function]}
                            placeholder="TITLE"
                            value=""
              />
              <textarea
                            cols="30"
                            onChange={[Function]}
                            placeholder="Description"
                            rows="4"
                            type="submit text"
                            value=""
              >
                            <input />
              </textarea>
              Category: 
              <select
                            className="add-habit-form-select form-control"
                            onChange={[Function]}
                            type="submit"
                            value=""
              >
                            <option
                                          value="Health/Fitness"
                            >
                                          Health/Fitness
                            </option>
                            <option
                                          value="Professional"
                            >
                                          Professional
                            </option>
                            <option
                                          value="Personal"
                            >
                                          Personal
                            </option>
              </select>
              <button
                            type="submit"
              >
                            Add Habit
              </button>
</form>,
          ],
          "className": "add-habit",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "add-habit-close-form",
              "onMouseDown": undefined,
              "src": "plus.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  maxLength="18"
                  onChange={[Function]}
                  placeholder="TITLE"
                  value=""
/>,
                <textarea
                  cols="30"
                  onChange={[Function]}
                  placeholder="Description"
                  rows="4"
                  type="submit text"
                  value=""
>
                  <input />
</textarea>,
                "Category: ",
                <select
                  className="add-habit-form-select form-control"
                  onChange={[Function]}
                  type="submit"
                  value=""
>
                  <option
                                    value="Health/Fitness"
                  >
                                    Health/Fitness
                  </option>
                  <option
                                    value="Professional"
                  >
                                    Professional
                  </option>
                  <option
                                    value="Personal"
                  >
                                    Personal
                  </option>
</select>,
                <button
                  type="submit"
>
                  Add Habit
</button>,
              ],
              "className": "add-habit-form",
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "maxLength": "18",
                  "onChange": [Function],
                  "placeholder": "TITLE",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <input />,
                  "cols": "30",
                  "onChange": [Function],
                  "placeholder": "Description",
                  "rows": "4",
                  "type": "submit text",
                  "value": "",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                "type": "textarea",
              },
              "Category: ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <option
                      value="Health/Fitness"
>
                      Health/Fitness
</option>,
                    <option
                      value="Professional"
>
                      Professional
</option>,
                    <option
                      value="Personal"
>
                      Personal
</option>,
                  ],
                  "className": "add-habit-form-select form-control",
                  "onChange": [Function],
                  "type": "submit",
                  "value": "",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Health/Fitness",
                      "value": "Health/Fitness",
                    },
                    "ref": null,
                    "rendered": "Health/Fitness",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Professional",
                      "value": "Professional",
                    },
                    "ref": null,
                    "rendered": "Professional",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Personal",
                      "value": "Personal",
                    },
                    "ref": null,
                    "rendered": "Personal",
                    "type": "option",
                  },
                ],
                "type": "select",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Add Habit",
                  "type": "submit",
                },
                "ref": null,
                "rendered": "Add Habit",
                "type": "button",
              },
            ],
            "type": "form",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "position": "bottom_right",
          "store": Object {
            "data": Object {
              "action": "",
              "message": "",
            },
            "error": [Function],
            "info": [Function],
            "success": [Function],
            "unwatch": [Function],
            "warning": [Function],
            "watch": [Function],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="add-habit"
>
            <img
                        className="add-habit-close-form"
                        onMouseDown={undefined}
                        src="plus.png"
            />
            <form
                        className="add-habit-form"
                        onSubmit={[Function]}
            >
                        <input
                                    maxLength="18"
                                    onChange={[Function]}
                                    placeholder="TITLE"
                                    value=""
                        />
                        <textarea
                                    cols="30"
                                    onChange={[Function]}
                                    placeholder="Description"
                                    rows="4"
                                    type="submit text"
                                    value=""
                        >
                                    <input />
                        </textarea>
                        Category: 
                        <select
                                    className="add-habit-form-select form-control"
                                    onChange={[Function]}
                                    type="submit"
                                    value=""
                        >
                                    <option
                                                value="Health/Fitness"
                                    >
                                                Health/Fitness
                                    </option>
                                    <option
                                                value="Professional"
                                    >
                                                Professional
                                    </option>
                                    <option
                                                value="Personal"
                                    >
                                                Personal
                                    </option>
                        </select>
                        <button
                                    type="submit"
                        >
                                    Add Habit
                        </button>
            </form>
</div>,
          <Container
            position="bottom_right"
            store={
                        Object {
                                    "data": Object {
                                      "action": "",
                                      "message": "",
                                    },
                                    "error": [Function],
                                    "info": [Function],
                                    "success": [Function],
                                    "unwatch": [Function],
                                    "warning": [Function],
                                    "watch": [Function],
                                  }
            }
/>,
        ],
        "style": Object {
          "height": "100vh",
          "width": "100vw",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                className="add-habit-close-form"
                onMouseDown={undefined}
                src="plus.png"
/>,
              <form
                className="add-habit-form"
                onSubmit={[Function]}
>
                <input
                                maxLength="18"
                                onChange={[Function]}
                                placeholder="TITLE"
                                value=""
                />
                <textarea
                                cols="30"
                                onChange={[Function]}
                                placeholder="Description"
                                rows="4"
                                type="submit text"
                                value=""
                >
                                <input />
                </textarea>
                Category: 
                <select
                                className="add-habit-form-select form-control"
                                onChange={[Function]}
                                type="submit"
                                value=""
                >
                                <option
                                                value="Health/Fitness"
                                >
                                                Health/Fitness
                                </option>
                                <option
                                                value="Professional"
                                >
                                                Professional
                                </option>
                                <option
                                                value="Personal"
                                >
                                                Personal
                                </option>
                </select>
                <button
                                type="submit"
                >
                                Add Habit
                </button>
</form>,
            ],
            "className": "add-habit",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "add-habit-close-form",
                "onMouseDown": undefined,
                "src": "plus.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    maxLength="18"
                    onChange={[Function]}
                    placeholder="TITLE"
                    value=""
/>,
                  <textarea
                    cols="30"
                    onChange={[Function]}
                    placeholder="Description"
                    rows="4"
                    type="submit text"
                    value=""
>
                    <input />
</textarea>,
                  "Category: ",
                  <select
                    className="add-habit-form-select form-control"
                    onChange={[Function]}
                    type="submit"
                    value=""
>
                    <option
                                        value="Health/Fitness"
                    >
                                        Health/Fitness
                    </option>
                    <option
                                        value="Professional"
                    >
                                        Professional
                    </option>
                    <option
                                        value="Personal"
                    >
                                        Personal
                    </option>
</select>,
                  <button
                    type="submit"
>
                    Add Habit
</button>,
                ],
                "className": "add-habit-form",
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "maxLength": "18",
                    "onChange": [Function],
                    "placeholder": "TITLE",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <input />,
                    "cols": "30",
                    "onChange": [Function],
                    "placeholder": "Description",
                    "rows": "4",
                    "type": "submit text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  "type": "textarea",
                },
                "Category: ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <option
                        value="Health/Fitness"
>
                        Health/Fitness
</option>,
                      <option
                        value="Professional"
>
                        Professional
</option>,
                      <option
                        value="Personal"
>
                        Personal
</option>,
                    ],
                    "className": "add-habit-form-select form-control",
                    "onChange": [Function],
                    "type": "submit",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Health/Fitness",
                        "value": "Health/Fitness",
                      },
                      "ref": null,
                      "rendered": "Health/Fitness",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Professional",
                        "value": "Professional",
                      },
                      "ref": null,
                      "rendered": "Professional",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Personal",
                        "value": "Personal",
                      },
                      "ref": null,
                      "rendered": "Personal",
                      "type": "option",
                    },
                  ],
                  "type": "select",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Add Habit",
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": "Add Habit",
                  "type": "button",
                },
              ],
              "type": "form",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "position": "bottom_right",
            "store": Object {
              "data": Object {
                "action": "",
                "message": "",
              },
              "error": [Function],
              "info": [Function],
              "success": [Function],
              "unwatch": [Function],
              "warning": [Function],
              "watch": [Function],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
